examples/generate_QG_time_series.py

```diff
@@ -0,0 +1,84 @@
+"""Truth-twin and observation time series for QG (quasi-geostrophic) model."""
+
+import numpy as np
+import scipy.ndimage.filters as filters
+from matplotlib import pyplot as plt
+from dapper.mods.QG import square
+import dapper.tools.progressbar as pb
+from dapper.mods.QG.sakov2008 import HMM
+from dapper.mods.QG.sakov2008 import obs_inds
+
+###########
+# Auxiliary plotting function
+###########
+
+def show(x0, psi=True, ax=None):
+    #
+    def psi_or_q(x):
+        return x if psi else compute_q(x)
+    #
+    if ax == None:
+        fig, ax = plt.subplots()
+
+    im = ax.imshow(psi_or_q(square(x0)))
+
+    if psi:
+        im.set_clim(-30, 30)
+    else:
+        im.set_clim(-28e4, 25e4)
+
+    def update(x):
+        im.set_data(psi_or_q(square(x)))
+    return update
+
+###########
+# Main
+###########
+
+## generate truth-twin and observation time series as per standard configuration
+## of Sakov2008, save the double time series
+xx, yy = HMM.simulate()
+
+# save data for easy re-use and debugging
+np.savez("QG_truth_obs_time_series.npz", xx=xx, yy=yy)
+
+# these lines can be used in place of above steps after first run,
+# comment out above
+tmp = np.load("QG_truth_obs_time_series.npz")
```

Comment:
`with` block is nicer than having to `close` the file later

Reply:
Prettier, and has other minor advantages.
https://www.pythonforbeginners.com/files/with-statement-in-python

Reply:
Ah, OK, I'll leave it with the close statement, but I have added an exception handler to open the saved data or generate it if it doesn't exist.